packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -1,7 +1,96 @@
+import "@paprika/helpers/lib/polyfills/elementClosest";
 import useListBox from "../../../useListBox";
 import invokeOnChange from "../../../helpers/invokeOnChange";
 import { KEY_PRESS, CLICK } from "../../../types";
 
+function getOptions(event) {
+  let list = null;
+  if (event.target.dataset.pkaAnchor && event.target.dataset.pkaAnchor === "list-filter-input") {
+    // the event is coming from the filter
+    list = event.target.closest("[data-pka-anchor='list-box-box']").querySelector("[data-pka-anchor='styled-list']");
+  } else {
+    list = event.target.querySelector("[data-pka-anchor='styled-list']");
+  }
+
+  // filtering only results with role="option" attribute
+  return [...list.children].filter(li => li.hasAttribute("role"));
```

Comment:
No need to `[...list.children]` if you `filter` because `filter` creates a new array anyway. Therefore `[...]` here is wasteful.

Reply:
list.children is a dom collection no an array, therefore lacks a filter implementation, requires to be turn into an array first.

Reply:
@nahumzs Why not `React.Children.toArray` instead? Would it be more explicit?

Reply:
ohh because are not React children are a Node `DOM` list, the children in the ListBox include ListBox.Trigger, ListBox.Footer, ListBox.Box etc. The `Node List` is just the nodes that have been rendered after ignoring any other kind of children.